Lab4/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -0,0 +1,64 @@
+package agh.ics.oop;
+
+import java.util.Objects;
+
+public class Animal {
+    private Vector2d location;
+    private MapDirection orientation;
+    private IWorldMap map;
+
+
+    public Animal(IWorldMap map) {
+        this.orientation = MapDirection.NORTH;
+        this.location = new Vector2d(2,2);
+        this.map = map;
+    }
+
+    public Animal(IWorldMap map, Vector2d initialPoistion) {
+        this(map);
```

Comment:
pomysł dobry, ale bardziej w drugą stronę - to mniejszy konstruktor powinien wołać większy, wtedy konstruktor z jednym parametrem sprowadzi się do `this(map, new Vector2d(2,2))`, a przypisywanie atrybutów będziemy mieli tylko w tym większym.